R/Simulations-class.R

```diff
@@ -91,54 +108,72 @@ GeneralSimulations <- function(data,
 ##'
 ##' @slot fit list with the final fits
 ##' @slot stopReasons list of stopping reasons for each simulation run
-##'
+##' @slot stop_report matrix of stopping rule outcomes
 ##' @export
 ##' @keywords classes
 .Simulations <-
-    setClass(Class="Simulations",
-             representation(fit="list",
-                            stopReasons="list"),
-             ## note: this prototype is put together with the prototype
-             ## for GeneralSimulations
-             prototype(fit=
-                           list(c(0.1, 0.2),
-                                c(0.1, 0.2)),
-                       stopReasons=
-                           list("A", "A")),
-             contains="GeneralSimulations",
-             validity=
-                 function(object){
-                     o <- Validate()
-
-                     nSims <- length(object@data)
-
-                     o$check(identical(length(object@fit), nSims),
-                             "fit must have same length as data")
-                     o$check(identical(length(object@stopReasons), nSims),
-                             "stopReasons must have same length as data")
-
-                     o$result()
-                 })
+  setClass(
+    Class = "Simulations",
+    representation(
+      fit = "list",
+      stopReasons = "list",
+      stop_report = "matrix"
+    ),
+    ## note: this prototype is put together with the prototype
+    ## for GeneralSimulations
+    prototype(
+      fit =
+        list(
+          c(0.1, 0.2),
+          c(0.1, 0.2)
+        ),
+      stopReasons =
+        list("A", "A"),
+      stop_report =
+        matrix(TRUE, TRUE)
```

Comment:
is it reasonable init for `stop_report`? It is the same as `stop_report = matrix(TRUE)`?